device/dell/x86_64-dellemc_s5296f_c3538-r0/plugins/sfputil.py

```diff
@@ -0,0 +1,545 @@
+        
+        # Absence of status throws error
+        if (reg_value == "" ):
+            return False
+
+        # Mask off bit for presence
+        mask = (1 << 1)
+        if (port_num > 96):
+            mask = (1 << 4)
+
+        # ModPrsL is active low
+        if reg_value & mask == 0:
+            return True
+
+        return False
+
+    def get_low_power_mode(self, port_num):
+
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+	# Port offset starts with 0x4000
+	port_offset = 16384 + ((port_num-1) * 16)
+
+	status = self.pci_get_value(self.BASE_RES_PATH, port_offset)
+	reg_value = int(status)
+
+        # Absence of status throws error
+        if (reg_value == "" ):
+            return False
+
+        # Mask off 4th bit for presence
+        mask = (1 << 6)
+
+        # LPMode is active high
+        if reg_value & mask == 0:
+            return False
+
+        return True
+
+    def set_low_power_mode(self, port_num, lpmode):
+
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+	# Port offset starts with 0x4000
+	port_offset = 16384 + ((port_num-1) * 16)
+
+	status = self.pci_get_value(self.BASE_RES_PATH, port_offset)
+	reg_value = int(status)
+
+        # Absence of status throws error
+        if (reg_value == "" ):
+            return False
+
+        # Mask off 4th bit for presence
+        mask = (1 << 6)
+		
+	# LPMode is active high; set or clear the bit accordingly
+        if lpmode is True:
+            reg_value = reg_value | mask
+        else:
+            reg_value = reg_value & ~mask
+
+        # Convert our register value back to a hex string and write back
+	status = self.pci_set_value(self.BASE_RES_PATH, reg_value, port_offset)
+
+        return True
+
+    def reset(self, port_num):
+
+	# Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
+
+	# Port offset starts with 0x4000
+	port_offset = 16384 + ((port_num-1) * 16)
+
+	status = self.pci_get_value(self.BASE_RES_PATH, port_offset)
+	reg_value = int(status)
+
+        # Absence of status throws error
+        if (reg_value == "" ):
+            return False
+
+        # Mask off 4th bit for presence
+        mask = (1 << 6)
+
+        # ResetL is active low
+        reg_value = reg_value & ~mask
+
+	# Convert our register value back to a hex string and write back
+	status = self.pci_set_value(self.BASE_RES_PATH, reg_value, port_offset)
+
+        # Sleep 1 second to allow it to settle
+        time.sleep(1)
+
+        reg_value = reg_value | mask
+
+	# Convert our register value back to a hex string and write back
+	status = self.pci_set_value(self.BASE_RES_PATH, reg_value, port_offset)
+
+        return True
+
+    def get_transceiver_change_event(self, timeout=0):
+        port_dict = {}
+        while True:
```

Comment:
There is no blocking operation to time out -- the timeout argument has no use in the function body, it is purely to satisfy the calling API.  This function runs forever, polling in a loop, with a 0.5 second poll interval -- it never returns to the caller.

Reply:
The point of the `timeout` argument is, if > 0, to allow for the function to return after a specified amount of time even if it doesn't detect a change event in that period of time. The function should run until there is a change event detected if `timeout` is 0, otherwise it should return after `timeout` milliseconds, returning `(True, {})`. See [here](https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_sfp/sfputilbase.py#L1271).

Reply:
@jleveque this has been addressed, please re-review